lib/instance_billing_flavor_check/utils.py

```diff
@@ -102,17 +101,36 @@ def get_rmt_ip_addr():
         return
 
     # use present RMT server IP first if registered
+    rmt_ips_addr = []
     for etc_hosts_line in etc_hosts_lines:
         if 'susecloud.net' in etc_hosts_line:
-            return etc_hosts_line.split('\t')[0]
-
-    if 'cloudregister' in sys.modules:
-        # get a new RMT IP
-        server = get_smt(False)
-        rmt_ips_addr = []
-        if server.get_ipv6():
-            rmt_ips_addr.append('[{}]'.format(server.get_ipv6()))
-        rmt_ips_addr.append(server.get_ipv4())
+            etc_hosts_addrs = etc_hosts_line.split('\t')[0]
+            # save all the IPs for susecloud.net
+            # as with IPv6 enabled there will be more than one line
+            rmt_ips_addr.append(etc_hosts_addrs.split(' ')[0])
```

Comment:
Any particular reason for doing two calls to split()? Wouldn't `etc_hosts_line.split()[0]` do the right thing already?

Also, you might want to filter out commented lines, as a line `# 1.2.3.4 foo.susecloud.net` would return `#` as IP address. Ideally, I'd avoid parsing /etc/hosts altogether and use a host lookup, but since the SMT host names differ by framework, that's a bit awkward too.

Reply:
> Any particular reason for doing two calls to split()? Wouldn't `etc_hosts_line.split()[0]` do the right thing already?
> 
Yes, it would, as it was originally but if `/etc/hosts` is fiddled manually it would not
For example,
```
# /etc/hosts
<registered_ip_from_registercloudguest> smt-ec2.susecloud.net   smt-ec2
<manually_added_ip> smt-ec2.susecloud.net smt-ec2
```
and with 
```
# utils.py
etc_hosts_addrs = etc_hosts_line.split('\t')[0]
print(etc_hosts_addrs)
print(etc_hosts_addrs.split(' ')[0])
```
would produce
```
ip-172-31-9-104:/home/ec2-user # ./instance-flavor-check 
<registered_ip_from_registercloudguest>
<registered_ip_from_registercloudguest>
<manually_added_ip> smt-ec2.susecloud.net smt-ec2
<manually_added_ip>
```

> Also, you might want to filter out commented lines, as a line `# 1.2.3.4 foo.susecloud.net` would return `#` as IP address. Ideally, I'd avoid parsing /etc/hosts altogether and use a host lookup, but since the SMT host names differ by framework, that's a bit awkward too.

I tested that again yesterday (as smoking test to double check all the changes works) and I noticed that
not sure how often it would happen but it wouldn't hurt to handle that
In case we didn't, it would return `['#', '<ip>']` or `['#']` if not registered/manually fiddled with
which either would make the request to `#` failed and try the next or just failed and report `BYOS`
IMO, it is better handling it (same way it is better to handle the other manual fiddling above)

Reply:
I'm not sure I follow. `split()` splits the given string into words, regardless whether they are separated by spaces or tabs, so I don't see why you would need to run it twice with explicit separators.

Reply:
You are right, I misread `split()` for `split(' ')` in your first comment